pipeline/testing/average_rating_test.py

```diff
@@ -0,0 +1,91 @@
+import sys
+import unittest
+
+from average_rating import calculate_analytics
+
+sys.path.insert(0, "/workspace/butterfly/pipeline/scripts/")
+
+
+class TestAverageRating(unittest.TestCase):
+    def test_calculate_analytics(self):
+        # Step 1: Initialize dependencies
+        matches = [
+            {
+                "community": "sds",
+                "from_user": "user1",
+                "generator": "commonLettersGenerator",
+                "match": "2022-07-25~IRbyUHik",
+                "timestamp": 1659195146351,
+                "users": ["user1", "user4"],
+                "value": 4,
+            },
+            {
+                "community": "sds",
+                "from_user": "user1",
+                "generator": "commonLettersGenerator",
+                "match": "2022-07-25~IRbyUHik",
+                "timestamp": 1659195148137,
+                "users": ["user1", "user4"],
+                "value": 4,
```

Comment:
Optional: There is an important case where one user might submit multiple ratings for the same match. In that case, we should only keep their latest rating.

In this case, the ratings are the same, so it does not affect the average. But if a user submitted ratings of 4 and 2 for the first match, then 2 for the second match, their average would become `8 / 3`, but really it should be `4 / 2` because the first two ratings are for the same match.

You could add another function that filters out ratings from the same user and match and only keep their most recent rating for each match.